Kernel/Thread.cpp

```diff
@@ -1234,6 +1234,13 @@ DispatchSignalResult Thread::dispatch_signal(u8 signal)
     auto signal_trampoline_addr = process.signal_trampoline().get();
     regs.set_ip(signal_trampoline_addr);
 
+#if ARCH(I386) || ARCH(X86_64)
+    // Userspace flags might be invalid for function entry, according to SYSV ABI.
+    // Set them to a known-good value to avoid weird handler misbehavior.
+    // Only IF (and the reserved bit 1) are set.
+    regs.set_flags(0x0202);
```

Comment:
Linux only clears TF, DF, and RF, apparently keeping the rest as-is. At least clearing the TF flag seems to have a noticeable effect (according to comments in the Linux kernel source code), since it is related to single-stepped execution. Is there a point in us doing something similar?